lib/github-status/support/github.rb

```diff
@@ -12,6 +12,11 @@ module GitHub
       def github
         @github ||= Octokit::Client.new access_token: access_token
       end
+
+      Contract None => String
+      def canonical_sha
+        @canonical_sha ||= (sha.match(/^.{40}$/) || github.commit(repo, sha).sha).to_s
```

Comment:
What happens if the commit doesn't exist on GitHub? Maybe the commit in the file is wrong or something. We'd be in a scenario where there's no local repository to fall back to, if that were the case. It may be correct to fail here, but a nice error could be helpful in that case.

Reply:
It looks like there's a reasonable response from the GH side.  I get this as the output:

> `GET https://api.github.com/repos/mook-as/concourse-github-status/commits/ab614f582: 422 - No commit found for SHA: ab614f582 // See: https://developer.github.com/v3/repos/commits/#get-a-single-commit`

I'm not sure having a local repo to fall back to makes any sense. If there's a file there, it _can't_ be a directory…

Reply:
Yes, we're on the same page there. If we're reading from a file, then we already know there's no repo we could fall back to.

What I was suggesting was that this:

```
ruby -r octokit -e "Octokit::Client.new(access_token: ENV['GITHUB_ACCESS_TOKEN']).commit('mook-as/concourse-github-status', 'ab614f582')"
Traceback (most recent call last):
	14: from -e:1:in `<main>'
	13: from /Users/colstrom/.gem/ruby/2.5.0/gems/octokit-4.13.0/lib/octokit/client/commits.rb:152:in `commit'
	12: from /Users/colstrom/.gem/ruby/2.5.0/gems/octokit-4.13.0/lib/octokit/connection.rb:19:in `get'
	11: from /Users/colstrom/.gem/ruby/2.5.0/gems/octokit-4.13.0/lib/octokit/connection.rb:156:in `request'
	10: from /Users/colstrom/.gem/ruby/2.5.0/gems/sawyer-0.8.1/lib/sawyer/agent.rb:94:in `call'
	 9: from /Users/colstrom/.gem/ruby/2.5.0/gems/faraday-0.15.4/lib/faraday/connection.rb:138:in `get'
	 8: from /Users/colstrom/.gem/ruby/2.5.0/gems/faraday-0.15.4/lib/faraday/connection.rb:387:in `run_request'
	 7: from /Users/colstrom/.gem/ruby/2.5.0/gems/faraday-0.15.4/lib/faraday/rack_builder.rb:143:in `build_response'
	 6: from /Users/colstrom/.gem/ruby/2.5.0/gems/faraday-0.15.4/lib/faraday/request/retry.rb:128:in `call'
	 5: from /Users/colstrom/.gem/ruby/2.5.0/gems/octokit-4.13.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
	 4: from /Users/colstrom/.gem/ruby/2.5.0/gems/octokit-4.13.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
	 3: from /Users/colstrom/.gem/ruby/2.5.0/gems/faraday-0.15.4/lib/faraday/response.rb:8:in `call'
	 2: from /Users/colstrom/.gem/ruby/2.5.0/gems/faraday-0.15.4/lib/faraday/response.rb:61:in `on_complete'
	 1: from /Users/colstrom/.gem/ruby/2.5.0/gems/faraday-0.15.4/lib/faraday/response.rb:9:in `block in call'
/Users/colstrom/.gem/ruby/2.5.0/gems/octokit-4.13.0/lib/octokit/response/raise_error.rb:16:in `on_complete': GET https://api.github.com/repos/mook-as/concourse-github-status/commits/ab614f582: 422 - No commit found for SHA: ab614f582 // See: https://developer.github.com/v3/repos/commits/#get-a-single-commit (Octokit::UnprocessableEntity)
```

could be handled better.

Reply:
Per offline discussion, this is a non-issue. These are already handled by `concourse-fuselage`.